tools/booster/Proxy.hs

```diff
@@ -300,11 +351,30 @@ respondEither mbStatsVar booster kore req = case req of
                             { state = simplifiedState
                             , nextStates = if null filteredNexts then Nothing else Just filteredNexts
                             }
-            pure result
+            let allLogs =
+                    if null filteredNexts
+                        then simplifiedStateLogs
+                        else combineLogs $ simplifiedStateLogs : filteredNextsLogs
+            pure $ appendLogs result allLogs
 
         isBottom :: ExecuteState -> Bool
         isBottom ExecuteState{term}
             | KoreJson.KJBottom _ <- term.term = True
         isBottom ExecuteState{predicate = Just p}
             | KoreJson.KJBottom _ <- p.term = True
         isBottom _ = False
+
+-- | Combine multiple, possibly empty/non-existent (Nothing) lists of logs into one
+combineLogs :: [Maybe [RPCLog.LogEntry]] -> Maybe [RPCLog.LogEntry]
+combineLogs logSources = case combineLogsImpl [] logSources of
+    [] -> Nothing
+    xs -> Just xs
+  where
+    combineLogsImpl :: [RPCLog.LogEntry] -> [Maybe [RPCLog.LogEntry]] -> [RPCLog.LogEntry]
+    combineLogsImpl acc = \case
+        [] -> acc
+        (Nothing : rest) -> combineLogsImpl acc rest
+        (Just logs : rest) -> combineLogsImpl (acc ++ logs) rest
+
+appendLogs :: ExecuteResult -> Maybe [RPCLog.LogEntry] -> ExecuteResult
+appendLogs ExecuteResult{..} newLogs = ExecuteResult{logs = combineLogs [logs, newLogs], ..}
```

Comment:
I am a bit scared when using `RecordWildCards`, it silently introduces variable bindings that may cause subtle bugs.
In this case, a record pun would work fine: `appendLogs res = res { logs =- combineLogs [rec.logs, newLogs] }`.
But we can change this "next time we touch the code" (i.e., today :) )

Reply:
Yes, I agree, let's get rid of `RecordWildCards` if possible. I could not find a better way to make the record update unambiguous, but I lack experience.